### PR TITLE
[RISCV] Recommit "Expand vp.stride.load to splat of a scalar load."

### PR DESCRIPTION
**Pull request by yetingk**
This is a recommit of #98140. The old commit should be rebased on #98205 which changes the feature of hardware zero stride optimization.

It's a similar patch as a214c521f8763b36dd400b89017f74ad5ae4b6c7 for vp.stride.load. Some targets prefer pattern (vmv.v.x (load)) instead of vlse with zero stride.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVCodeGenPrepare.cpp [hunk 1]
```diff
@@ -155,6 +163,47 @@ bool RISCVCodeGenPrepare::visitIntrinsicInst(IntrinsicInst &I) {
   return true;
 }
 
+bool RISCVCodeGenPrepare::expandVPStrideLoad(IntrinsicInst &II) {
+  if (ST->hasOptimizedZeroStrideLoad())
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVCodeGenPrepare.cpp:167]: We don't have test coverage for this?

Comment (lukel97) [reply llvm/lib/Target/RISCV/RISCVCodeGenPrepare.cpp:167]: I think we need to negate this condition now

Comment (lukel97) [reply llvm/lib/Target/RISCV/RISCVCodeGenPrepare.cpp]: Oops ignore my previous comment, I got confused

File: llvm/test/CodeGen/RISCV/rvv/strided-vpload.ll
Comment (lukel97) [inline llvm/test/CodeGen/RISCV/rvv/strided-vpload.ll:1]: Should we still add a RUN line with `optimized-zero-stride-load` to ensure that it gets 'unexpanded' back into a zero strided load during instruction selection?

**Review by lukel97 — APPROVED** [review]
LGTM

**Review by wangpc-pp — APPROVED** [review]

**Pull request merged**